### DISCUSSION TITLE
OpenOCD Connection Error Possibly Caused by Usage of k_msleep Function

### DISCUSSION THREAD

**Discussion by FlameKm** [zephyrproject-rtos/zephyr#61474] [Q&A]
I was burning stm32f103c8t6, ​​and after burning once, I found that there is no SWD configured, and the OPENOCD cannot be connected under normal operation. Only when the OPENOCD is connected at the moment of reset, can the burning be performed.
I found a way to use CONFIG_GPIO_STM32_SWJ_DISABLE, but I can't use it after configuring
Next I checked the docs and found this
![企业微信截图_16920168639288](https://github.com/zephyrproject-rtos/zephyr/assets/119095160/315b9cdb-1587-4145-86b2-a0a1ecc4193d)
Here I found that there is no relevant description information in the device tree "st, stm32-gpio.yaml" file, and I have no choice
Hoping to provide me with a way to configure swd in the device tree

The environment I use is: 
Board: stm32f103c8t6 
Board file: stm32f103_mini 
Burning tool: OPENOCD+DAP

**Comment by FlameKm** [comment]
I have repositioned the problem. The issue lies in the usage of `k_msleep`. After adding this line of code, I encounter an inability to reprogram the device a second time. I need to press the reset button and only then, during the split second of initiating the reprogramming, am I able to proceed. This problem doesn't occur when using other kernel functions. Clearly, the issue is not related to enabling SWJ as mentioned above.

**Reply by GeorgeCGV** [reply]
SWJ can be configured in DTS within pinctrl: https://github.com/zephyrproject-rtos/zephyr/blob/77698e8a2235f575a64a58d19b3205f6e0c6b85b/dts/bindings/pinctrl/st%2Cstm32f1-pinctrl.yaml#L20-L39

You might have a look into #56096.

**Reply by FlameKm** [reply]
Thank you very much, I solved the problem that I had configured the swd, but still didn't solve my inability to burn. A day ago, though, suddenly it worked, and I still haven't found out what caused it, but the burn prompts as
```shell
Warn : could not read product string for device 0x0d28:0x0204: Operation timed out
```
After a while it will go to normal burn state No cause of the problem has been found, but now there is no longer a problem of not being able to burn.
